### PR TITLE
fix build failed and support multiple outputs

### PR DESCRIPTION
**Pull request by fumihwh**
#53 
Finally, I fixed all build errors and passed the CI test.
But I am not sure if the fix is optimized.
And I added multiple outputs support.

### CODE REVIEW

File: onnx_tf/backend.py [hunk 1]
```diff
@@ -1079,7 +1085,7 @@ def handle_split(cls, node, input_dict):
     split = (tf.constant(node.attrs["split"]) if
              "split" in node.attrs else input_dict[node.inputs[1]])
     axis = node.attrs["axis"]
-    return [tf.split(input_dict[node.inputs[0]], split, axis)]
+    return [el for el in tf.split(input_dict[node.inputs[0]], split, axis)]
```
Comment (tjingrant) [inline onnx_tf/backend.py]: This looks strange.. is it possible to do `list(tf.split(...))` instead?

File: onnx_tf/backend.py [hunk 2]
```diff
@@ -1102,6 +1108,12 @@ def handle_thresholded_relu(cls, node, input_dict):
     epsilon = 1e-5
     return [tf.nn.relu(x) - tf.nn.relu(tf.sign(alpha - x + epsilon) * x)]
 
+  @classmethod
+  def handle_top_k(cls, node, input_dict):
+    x = input_dict[node.inputs[0]]
+    k = node.attrs["k"] if "k" in node.attrs else 1
+    return [el for el in tf.nn.top_k(x, k=k)]
```
Comment (tjingrant) [inline onnx_tf/backend.py]: Same as above. Does `list` work?

File: test/backend/test_node.py [hunk 3]
```diff
@@ -510,11 +510,11 @@ def test_slice(self):
 
   def test_split(self):
     split = [3, 3, 4]
-    node_def = helper.make_node("Split", ["X"], ["Z"], axis=0, split=split)
+    node_def = helper.make_node("Split", ["X"], ["Z%i" % i for i in range(len(split))], axis=0, split=split)
     x = self._get_rnd([100]).reshape([10, 10])
 
     output = run_node(node_def, [x])
-    for a, b in zip(output["Z"], np.split(x,np.cumsum(split))[:-1]):
+    for a, b in zip([el for el in output], np.split(x,np.cumsum(split))[:-1]):
```
Comment (tjingrant) [inline test/backend/test_node.py]: Same. Would `list` work?

File: onnx_tf/backend.py [hunk 4]
```diff
@@ -585,12 +577,22 @@ def _pool(cls, node, input_dict, pool_func, can_pad_zero):
 
     # By default, do not pad
     pad = "VALID"
-
-    if "pads" in node.attrs.keys():
+    pads = node.attrs["pads"] if "pads" in node.attrs else None
+
+    # Pooling1D
```
Comment (tjingrant) [inline onnx_tf/backend.py]: Why do we have to special case 1-dimensional pooling?
As per TF documentation here (https://www.tensorflow.org/api_docs/python/tf/nn/pool), pooling op does support 1d pooling. 

Quote from the documentation:
> For N=1, the valid values are "NWC" (default) and "NCW".

File: test/frontend/test_node.py [hunk 5]
```diff
@@ -70,7 +70,11 @@ def do_test_expected(self):
     onnx_graph = convert_graph(tf_graph, output_node)
     onnx_model = helper.make_model(onnx_graph)
     backend_rep = prepare(onnx_model)
-    backend_output = backend_rep.run(onnx_feed_dict)[output_name]
+    backend_output = []
+    for ext_output in backend_rep.predict_net.external_output:
+      backend_output.append(backend_rep.run(onnx_feed_dict)[ext_output])
```
Comment (tjingrant) [inline test/frontend/test_node.py]: Looks like if we have N outputs, the test will run N times.
Any better way to do this?

File: onnx_tf/backend.py [hunk 6]
```diff
@@ -390,7 +382,7 @@ def onnx_graph_to_tensorflow_net(cls, graph_def):
     # defined tensors so we can have access to the placeholders
     # to feed in input tensors when we run the graph.
     original_input_dict = dict(input_dict_items)
-    output_dict = dict()
+    output_dict = input_dict
```
Comment (tjingrant) [inline onnx_tf/backend.py]: Is this necessary?
Does not look very reassuring...

Comment (fumihwh) [reply onnx_tf/backend.py]: @tjingrant 
For `test_operator_chunk`, no node in graph and input is just output. Just for that case.

File: onnx_tf/backend.py (hunk 4 above)
Comment (fumihwh) [reply onnx_tf/backend.py]: `1D` means `NWC` or `NCW`

File: onnx_tf/backend.py [hunk 7]
```diff
@@ -1085,7 +1085,7 @@ def handle_split(cls, node, input_dict):
     split = (tf.constant(node.attrs["split"]) if
              "split" in node.attrs else input_dict[node.inputs[1]])
     axis = node.attrs["axis"]
-    return [el for el in tf.split(input_dict[node.inputs[0]], split, axis)]
+    return [list(tf.split(input_dict[node.inputs[0]], split, axis))]
```
Comment (tjingrant) [inline onnx_tf/backend.py]: I think we shouldn't need the outer brackets. It should simply be:
`list(tf.split(input_dict[node.inputs[0]], split, axis))`

File: onnx_tf/backend.py (hunk 4 above)
Comment (fumihwh) [reply onnx_tf/backend.py]: @tjingrant 
For reference, 
https://github.com/tensorflow/tensorflow/blob/7c324811f10eaf049735ace7b6a228ccc931d8ff/tensorflow/python/layers/pooling.py#L32

**Comment by fumihwh** [comment]
@tjingrant 
`_compatibility_avg_pool` supports 2D (`NCHW` or `NHWC`) only now.
And the implement is very hard coding to fit the 2D.
How should us deal with this?

**Comment by tjingrant** [comment]
@fumihwh I wrote this one to cover an important test case. This shouldn't be used by the user anyways... Let's leave it there and I'll deal with it later.

**Comment by tjingrant** [comment]
Thanks a lot for your help! Glad to see a passing CI badge again!
